File: CASE_Simple_Example_V4/Hardened/HAMR/hamr/ext-c/GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor/GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor.c
#include <GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor_api.h>
#include <GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor.h>
#include <ext.h>

// This file will not be overwritten so is safe to edit

Unit HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor_initialise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor.c", "", "HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor_initialise_", 0);

  // examples of api setter and logging usage

  DeclNewHAMR_SW_Mission(t0);
  HAMR_SW_Mission_example(SF &t0);
  api_put_FlightPlan_Out__HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor(SF &t0);

  api_put_Alert__HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor(SF_LAST);

  api_logInfo__HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor(SF string("Example logInfo"));

  api_logDebug__HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor(SF string("Example logDebug"));

  api_logError__HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor(SF string("Example logError"));
}

Unit HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor_finalise_(STACK_FRAME_ONLY) {
  DeclNewStackFrame(caller, "GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor.c", "", "HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor_finalise_", 0);
}

Unit HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor_handle_FlightPlan_In_(
  STACK_FRAME
  HAMR_SW_Mission value) {
  DeclNewStackFrame(caller, "GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor.c", "", "HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor_handle_FlightPlan_In_", 0);

  DeclNewString(FlightPlan_InString);
  String__append(SF (String) &FlightPlan_InString, string("HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor_handle_FlightPlan_In called"));
  api_logInfo__HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor (SF (String) &FlightPlan_InString);

  DeclNewString(_str);
  String__append(SF (String) &_str, string("Received on FlightPlan_In: "));
  HAMR_SW_Mission_string_(SF (String) &_str, value);
  api_logInfo__HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor(SF (String) &_str);

  // examples of api getter usage

  DeclNewHAMR_SW_MapArray(t0);
  if(api_get_FlyZones__HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor(SF &t0)) {
    DeclNewString(FlyZones_str);
    String__append(SF (String) &FlyZones_str, string("Received on FlyZones: "));
    HAMR_SW_MapArray_string_(SF (String) &FlyZones_str, &t0);
    api_logInfo__HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor(SF (String) &FlyZones_str);
  }

  DeclNewHAMR_SW_Mission(t1);
  if(api_get_FlightPlan_In__HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor(SF &t1)) {
    DeclNewString(FlightPlan_In_str);
    String__append(SF (String) &FlightPlan_In_str, string("Received on FlightPlan_In: "));
    HAMR_SW_Mission_string_(SF (String) &FlightPlan_In_str, &t1);
    api_logInfo__HAMR_SW_GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor(SF (String) &FlightPlan_In_str);
  }
}

File: CASE_Simple_Example_V4/Hardened/HAMR/src/main/component/HAMR/SW/GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor.scala
// #Sireum

package HAMR.SW

import org.sireum._
import HAMR._

// This file will not be overwritten so is safe to edit
object GeofenceMonitor_Impl_SW_GeofenceMonitor_GeofenceMonitor {

  def initialise(api: GeofenceMonitor_Impl_Initialization_Api): Unit = {
    // example api usage

    api.logInfo("Example info logging")
    api.logDebug("Example debug logging")
    api.logError("Example error logging")

    api.put_FlightPlan_Out(SW.Mission.example())
    api.put_Alert()
  }

  def handle_FlightPlan_In(api: GeofenceMonitor_Impl_Operational_Api, value : SW.Mission): Unit = {
    api.logInfo("example handle_FlightPlan_In implementation")
    api.logInfo(s"received ${value}")
    // example api usage

    val apiUsage_FlyZones: Option[SW.MapArray] = api.get_FlyZones()
    api.logInfo(s"Received on FlyZones: ${apiUsage_FlyZones}")
    val apiUsage_FlightPlan_In: Option[SW.Mission] = api.get_FlightPlan_In()
    api.logInfo(s"Received on FlightPlan_In: ${apiUsage_FlightPlan_In}")
  }

  def activate(api: GeofenceMonitor_Impl_Operational_Api): Unit = { }

  def deactivate(api: GeofenceMonitor_Impl_Operational_Api): Unit = { }

  def finalise(api: GeofenceMonitor_Impl_Operational_Api): Unit = { }

  def recover(api: GeofenceMonitor_Impl_Operational_Api): Unit = { }
}
